src/components/header/Header.tsx

```diff
@@ -30,28 +39,44 @@ const HeaderContainer = styled.div`
     filter: opacity(0.7);
   }
 `;
+
+const StyledSearchButton = styled(IconButton)<Active>`
+  background-color: ${({ theme, isActive }) =>
+    isActive ? theme.colors.dark : 'transparent'};
+  .button-icon {
+    color: ${({ theme, isActive }) =>
+      isActive ? theme.colors.light : theme.colors.dark};
+  }
+`;
+
```

Comment:
I feel like this behaviour should be baked in to the IconButton. It could be generalised to support a color prop too like 
```
  background-color: ${({ theme, isActive, color }) =>
    isActive ? theme.palette[color].contrast : 'transparent'};
  .button-icon {
    color: ${({ theme, isActive }) =>
      isActive ? theme.palette[color].main : theme.palette[color].contrast};
  }
```